### PR TITLE
Replace -lpthread with -pthread

### LINKED ISSUE #1401: riscv64 FTBFS due to atomics
Some time ago, I uploaded 5.1.0 to Debian's unstable. It successfully built on all Linux & kFreeBSD architectures with one exception: riscv64 (which is a new and not official Debian arch yet).

5.1.0 included the commit 749caf14ae73a9ab1c48e538a8af09addbb35ee7, so I was kind of expecting it to work, however it seems to have failed with this:
```
[...]
checking whether C11 atomics is compilable... yes
checking whether GCC __atomic atomics is compilable... yes
checking whether GCC __sync atomics is compilable... yes
[...]
gcc  -o test/integration/extent test/integration/extent.o test/src/btalloc.integration.o test/src/btalloc_0.integration.o test/src/btalloc_1.integration.o test/src/math.integration.o test/src/mtx.integration.o test/src/mq.integration.o test/src/SFMT.integration.o test/src/test.integration.o test/src/thd.integration.o test/src/timer.integration.o src/nstime.integration.o src/malloc_io.integration.o -Wl,-rpath,lib lib/libjemalloc.so -Wl,-z,relro -lstdc++ -lpthread -lm 
lib/libjemalloc.so: undefined reference to `__atomic_compare_exchange_1'
collect2: error: ld returned 1 exit status
```

Looking at similar bug reports in Debian's BTS and the web about this, I've found [a bug report against webkit2gtk](https://bugs.debian.org/895969) where the reporter said:
> This happens due to the fact that riscv64 has native atomics support, but only for 32bit and 64bit operations. Support for 8bit/16bit atomic operations is provided by libatomic, but webkit2gtk doesn't link against libatomic, so the build fails.
>
> There is support for automatically linking in libatomic when required, but for that the linker needs to be called with the "-pthread" parameter (not "-lpthread").

Another [one on Facebook's zstd](https://bugs.debian.org/895259) resulted into a merged PR, facebook/zstd#1103, that did the same (`-lpthread` -> `-pthread`). I suppose an alternative would be to add `-latomic` directly?

I'd send a PR, but I'm a bit hesitant given all the past complexities around atomics, plus all the potential for breakage in corner cases I'm not familiar with and unable to test (e.g. would this work with Clang?).

Would love to hear your thoughts! I set up a working riscv64 environment (using qemu-user-static) and would be able to test any proposed fixes here :)

### PR DESCRIPTION
**Pull request by paravoid**
This automatically adds -latomic if and when needed, e.g. on riscv64
systems.

Fixes #1401.

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -420,7 +420,7 @@ $(objroot)test/unit/%$(EXE): $(objroot)test/unit/%.$(O) $(C_JET_OBJS) $(C_TESTLI
 
 $(objroot)test/integration/%$(EXE): $(objroot)test/integration/%.$(O) $(C_TESTLIB_INTEGRATION_OBJS) $(C_UTIL_INTEGRATION_OBJS) $(objroot)lib/$(LIBJEMALLOC).$(IMPORTLIB)
 	@mkdir -p $(@D)
-	$(CC) $(TEST_LD_MODE) $(LDTARGET) $(filter %.$(O),$^) $(call RPATH,$(objroot)lib) $(LJEMALLOC) $(LDFLAGS) $(filter-out -lm,$(filter -lrt -lpthread -lstdc++,$(LIBS))) $(LM) $(EXTRA_LDFLAGS)
+	$(CC) $(TEST_LD_MODE) $(LDTARGET) $(filter %.$(O),$^) $(call RPATH,$(objroot)lib) $(LJEMALLOC) $(LDFLAGS) $(filter-out -lm,$(filter -pthread -lrt -lstdc++,$(LIBS))) $(LM) $(EXTRA_LDFLAGS)
```
Comment (interwq) [inline Makefile.in]: Any particular reason `-pthread` is placed before lrt? If not let's keep the original ordering which might still matter on some platforms.

Comment (paravoid) [reply Makefile.in]: Alright, done!

Comment (interwq) [reply Makefile.in]: Thanks for the patch @paravoid! Will merge soon.
